WebApi/Controllers/BrandsController.cs

```diff
@@ -0,0 +1,81 @@
+﻿using EcommerceStore.Application.Interfaces.Services;
+using EcommerceStore.Infrastucture.Persistence.Models.InputModels;
+using EcommerceStore.Infrastucture.Persistence.Models.ViewModels;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace EcommerceStore.WebApi.Controllers
+{
+    [ApiController]
+    [Route("api/brands")]
+    public class BrandsController : ControllerBase
+    {
+        private readonly IBrandService _brandService;
+
+        public BrandsController(IBrandService brandService)
+        {
+            _brandService = brandService;
+        }
+
+        [HttpGet("{brandId}")]
+        public async Task<ActionResult<BrandViewModel>> GetByIdAsync([FromRoute] int brandId)
+        {
+            BrandViewModel brandViewModel = await _brandService.GetByIdAsync(brandId);
+
+            if (brandViewModel == null)
```

Comment:
It is better to throw ex from service with the meaningful error message rather then returning notFound
you can create custom exception class e.g. (think of better name)
```
public class CustomException : Exception
    {
        public List<string> ErrorMessages { get; private set; }
        public Dictionary<string, List<string>> FieldSpecificMessages { get; private set; }

        public CustomException(List<string> messages)
        {
            ErrorMessages = messages;
        }

        public ValidationException(string message)
        {
            ErrorMessages = new List<string> { message };
        }

public CustomException(ModelStateDictionary modelState)
        {
            Dictionary<string, string[]> extractedErrors = new();

            FieldSpecificMessages = modelState.Keys.ToDictionary(k => k, k => modelState[k].Errors.Select(e => e.ErrorMessage).ToList());
            GeneralMessages = new List<string>();
        }
}
```
And you can define ExceptionMessages class and throw this custom exception with meaningful description defined in ExceptionMessagesClass(it will just contain a set of static strings for different error messages)

Reply:
In this way it will be easier for client app to know the format of exceptions. you can throw this exception from controller as well if model state is invalid